Error setting profile photo

If you upload a photo, go to the set profile picture page, and then try to set it as profile picture, it fails...

```
Internal Server Error (500) for request GET /photos/setProfilePhoto?submit=Set
Execution exception (In /app/controllers/Photos.java around line 109)
IllegalArgumentException occured : id to load is required for loading

play.exceptions.JavaExecutionException: id to load is required for loading
       at play.mvc.ActionInvoker.invoke(ActionInvoker.java:231)
       at Invocation.HTTP Request(Play!)
Caused by: java.lang.IllegalArgumentException: id to load is required for loading
       at org.hibernate.event.LoadEvent.<init>(LoadEvent.java:89)
       at org.hibernate.event.LoadEvent.<init>(LoadEvent.java:61)
       at org.hibernate.impl.SessionImpl.get(SessionImpl.java:1002)
       at org.hibernate.impl.SessionImpl.get(SessionImpl.java:998)
       at org.hibernate.ejb.AbstractEntityManagerImpl.find(AbstractEntityManagerImpl.java:614)
       at org.hibernate.ejb.AbstractEntityManagerImpl.find(AbstractEntityManagerImpl.java:589)
       at play.db.jpa.JPQL.findById(JPQL.java:36)
       at models.Photo.findById(Photo.java)
       at controllers.Photos.setProfilePhoto(Photos.java:109)
       at play.mvc.ActionInvoker.invokeWithContinuation(ActionInvoker.java:548)
       at play.mvc.ActionInvoker.invoke(ActionInvoker.java:502)
       at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:478)
       at play.mvc.ActionInvoker.invokeControllerMethod(ActionInvoker.java:473)
       at play.mvc.ActionInvoker.invoke(ActionInvoker.java:161)
       ... 1 more
```

--- a/app/views/Photos/setProfilePhotoPage.html
+++ b/app/views/Photos/setProfilePhotoPage.html
@@ -35,14 +35,9 @@
 
   <ul class="photo-list">
     #{list items:photos, as:'photo' }
-      <li>
         <img src="@{Thumbnails.get120x120(photo.id)}" />
-        #{if currentUser.equals(photo.owner)}
-          #{form @Photos.setProfilePhoto(photo.id)}
-            <input type="submit" name="submit" class="btn" value=&{'photo.setAsProfilePhoto'} />
-          #{/form}
-        #{/if}
-      </li>
+        <a class = "button" href="@{Photos.setProfilePhoto(photo.id)}">
+        &{'photo.setAsProfilePhoto'}</a> <br>
     #{/list}
   </ul>
 </div>

--- a/app/views/tags/post.html
+++ b/app/views/tags/post.html
@@ -17,7 +17,9 @@
             <iframe src="${((models.Link)_item).external_url}"></iframe>
           </div>
         #{/if}
+        #{if _item.getClass().getName() != "models.Photo"}
         ${_item.content.markdown().nl2br()}
+        #{/if}
       </div>  
     </div>
 

